src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BasePatternSwitchLocalRewriter.cs

```diff
@@ -145,6 +145,18 @@ public bool MightAssignSomething(BoundExpression expr)
                     return this._mightAssignSomething;
                 }
 
+                public override BoundNode VisitReturnStatement(BoundReturnStatement node)
+                {
+                    if (node.RefKind != RefKind.None)
+                    {
+                        // might be taking a ref to a variable in a lambda that the caller assigns.
+                        this._mightAssignSomething = true;
```

Comment:
I don't see how this can be hit without either an error about ref returning a variable outside its lifetime, or an error on the call to such a method with a ref parameter and one of the variables we are tracking.

Reply:
Agreed; will remove the return statement visitor.

---
In reply to: [180583839](https://github.com/dotnet/roslyn/pull/25963#discussion_r180583839) [](ancestors = 180583839)